subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -386,6 +425,123 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
 					sizeof(struct wifi_status));
 }
 
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_ROAMING
+static int wifi_start_roaming(uint32_t mgmt_request, struct net_if *iface,
+			      void *data, size_t len)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+
+	if (roaming_params.is_11r_used) {
+		if (wifi_mgmt_api == NULL ||
+		    wifi_mgmt_api->start_11r_roaming == NULL) {
+			return -ENOTSUP;
+		}
+
+		return wifi_mgmt_api->start_11r_roaming(dev);
+	} else if (roaming_params.is_11k_enabled) {
+		memset(&roaming_params.neighbor_rep, 0x0, sizeof(roaming_params.neighbor_rep));
+		if (wifi_mgmt_api == NULL
+		    || wifi_mgmt_api->send_11k_neighbor_request == NULL) {
+			return -ENOTSUP;
+		}
+
+		return wifi_mgmt_api->send_11k_neighbor_request(dev, NULL);
+	} else if (wifi_mgmt_api == NULL || wifi_mgmt_api->btm_query == NULL) {
+		return -ENOTSUP;
+	}
+
+	return wifi_mgmt_api->btm_query(dev, 0x10);
+}
+
+NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_START_ROAMING, wifi_start_roaming);
+
+static int wifi_neighbor_rep_complete(uint32_t mgmt_request, struct net_if *iface,
+				      void *data, size_t len)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+	struct wifi_scan_params params = {0};
+	int i = 0;
```

Comment:
Unnecessary init here. You could replace this and the for loop by
```
for (int i = 0; i < roaming....) {
```

Reply:
Updated.